views/ViewsData.js

```diff
@@ -518,3 +518,34 @@ function log(stuff) {
     }
   }, 500)
 }
+
+export function ListItemDataProvider(props) {
+  let value = useMemo(() => ({ [props.context]: props.item }), [
+    props.context,
+    props.item,
+  ])
+  let valueItem = useMemo(() => ({
+    [`${props.context}_item`]: {
+      index: props.index,
+      indexReverse: props.list.length - props.index,
+      isFirst: props.index === 0,
+      isLast: props.index === props.list.length - 1,
+    },
+  }))
+
+  return (
+    <DataProvider
+      context={props.context}
+      value={value}
+      viewPath={props.viewPath}
+    >
+      <DataProvider
+        context={`${props.context}_item`}
+        value={valueItem}
+        viewPath={props.viewPath}
+      >
+        {props.children}
+      </DataProvider>
+    </DataProvider>
+  )
+}
```

Comment:
leaving this comment here because depending on how the convo on onSubmit/onChange goes we may need to do something different here to allow for that function to be defined or passed